From ca16b59cfade15c80b2fd6558891f6c905a2db33 Mon Sep 17 00:00:00 2001
From: G Kranthi <gudishax.kranthikumar@intel.com>
Date: Thu, 14 Jul 2016 12:28:39 +0530
Subject: [PATCH 2128/2508] ASoC: Intel: Skylake: Fix loadable library
 structure in manifest

commit 440aa2d286d7264b821fb7137679b9bdde9c0307 from
https://github.com/01org/linux-apollolake-i

This patch will assign library info as a scalar value
instead of pointer, which simplifies parsing manifest
information.

Change-Id: I0e4bc81d9fb8da31896bde3bb40b0460babd996c
Signed-off-by: G Kranthi <gudishax.kranthikumar@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/10215
Reviewed-by: B, Jayachandran <jayachandran.b@intel.com>
Tested-by: Jayanti, Satya Charitardha <satya.charitardha.jayanti@intel.com>
---
 sound/soc/intel/skylake/skl-topology.c       |   25 +------------------------
 sound/soc/intel/skylake/skl-tplg-interface.h |    4 ++--
 2 files changed, 3 insertions(+), 26 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-topology.c b/sound/soc/intel/skylake/skl-topology.c
index 71382d3..a2b2133 100644
--- a/sound/soc/intel/skylake/skl-topology.c
+++ b/sound/soc/intel/skylake/skl-topology.c
@@ -2462,18 +2462,7 @@ static int skl_manifest_load(struct snd_soc_component *cmpnt,
 
 	minfo = &skl->skl_sst->manifest;
 	mdata = manifest->priv.data;
-
-	memcpy(&minfo->cfg.mem_sts, mdata, (sizeof(minfo->cfg.mem_sts)));
-	mdata += sizeof(struct skl_mem_status);
-	memcpy(&minfo->cfg.slw_frq, mdata, (sizeof(minfo->cfg.slw_frq)));
-	mdata += sizeof(struct skl_dsp_freq);
-	memcpy(&minfo->cfg.fst_frq, mdata, (sizeof(minfo->cfg.fst_frq)));
-	mdata += sizeof(struct skl_dsp_freq);
-	memcpy(&minfo->cfg.dmacfg, mdata, (sizeof(minfo->cfg.dmacfg)));
-	mdata += sizeof(struct skl_dma_buff_cfg);
-
-	minfo->lib_count = *mdata;
-
+	memcpy(minfo, mdata, sizeof(*minfo));
 	if (minfo->lib_count > HDA_MAX_LIB) {
 		dev_err(bus->dev, "Exceeding max Library count. Got:%d\n",
 				minfo->lib_count);
@@ -2481,18 +2470,6 @@ static int skl_manifest_load(struct snd_soc_component *cmpnt,
 		goto exit_manifest;
 	}
 
-	mdata++;
-	minfo->lib = devm_kzalloc(bus->dev, minfo->lib_count *
-			(sizeof(struct lib_info)), GFP_KERNEL);
-
-	if (!minfo->lib) {
-		ret = -ENOMEM;
-		goto exit_manifest;
-	}
-
-	memcpy(minfo->lib, mdata, (u32) (minfo->lib_count *
-				(sizeof(struct lib_info))));
-
 exit_manifest:
 	return ret;
 }
diff --git a/sound/soc/intel/skylake/skl-tplg-interface.h b/sound/soc/intel/skylake/skl-tplg-interface.h
index 938084c..cbfdf30 100644
--- a/sound/soc/intel/skylake/skl-tplg-interface.h
+++ b/sound/soc/intel/skylake/skl-tplg-interface.h
@@ -34,7 +34,7 @@
 #define SKL_MOD_NAME 40 /* Length of GUID string */
 #define STEREO 2
 
-#define LIB_NAME_LENGTH 512
+#define LIB_NAME_LENGTH 128
 #define HDA_MAX_LIB    16
 #define MAX_DMA_CFG    24
 #define SKL_UUID_STR_SZ 40
@@ -309,7 +309,7 @@ struct skl_dfw_manifest {
 
 	/* library info */
 	u8 lib_count;
-	struct lib_info *lib;
+	struct lib_info lib[HDA_MAX_LIB];
 
 } __packed;
 
-- 
1.7.5.4

